firmware/defmt-itm/src/lib.rs

```diff
@@ -0,0 +1,76 @@
+//! [`defmt`](https://github.com/knurling-rs/defmt) global logger over RTT.
+//!
+//! NOTE when using this crate it's not possible to use (link to) the `rtt-target` crate
+//!
+//! To use this crate, link to it by importing it somewhere in your project.
+//!
+//! ```
+//! // src/main.rs or src/bin/my-app.rs
+//! use defmt_rtt as _;
+//! ```
+
+#![no_std]
+
+use core::{
+    ptr::NonNull,
+    sync::atomic::{AtomicBool, Ordering},
+};
+
+use cortex_m::{interrupt, itm, peripheral::ITM, register};
+
+static ENABLED: AtomicBool = AtomicBool::new(false);
+
+/// Enables defmt logging over the ITM stimulus port 0
+pub fn enable(itm: ITM) {
```

Comment:
we may want to make the stimulus port configurable in the feature that can be added backwards compatibility by adding a second function with signature `fn(ITM, port_number: u8)`